### PR TITLE
Avoid building shared libraries + support all upstream-supported platforms

### PR DESCRIPTION
**Pull request by leofang**
xref: https://github.com/conda-forge/cutlass-feedstock/pull/19#issuecomment-2106557517

Let's get this feedstock unblocked by restoring what CUTLASS is meant to be: A header-only C++ template library. It is too much to build a shared library. It's useless and does not serve for runtime JIT compilation needs. This feedstock should be as simple as [cuda-cccl-feedstock](https://github.com/conda-forge/cuda-cccl-feedstock) and [cccl-feedstock](https://github.com/conda-forge/cccl-feedstock) (CCCL is also a header-only lib).

I also build support for all Linux & Windows platforms to pave the way for an upcoming NVIDIA release.

I add myself as a maintainer.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
@h-vetinari this is ready. I spoke with the CUTLASS PM @mnicely, and we are unaware of any libraries that use `libcutlass.so`. This PR would make things a lot simpler and unblock some of the PRs I am preparing. Could you offer your feedback when you get a chance please (I know you're busy with stdlib & numpy 2 infra changes)? Thanks! 🙂

**Comment by leofang** [comment]
(FYI there's also [CUTLASS wheels](https://pypi.org/project/nvidia-cutlass/#files) are they are lightweight too. The CUTLASS team adds Python support in the wheel, once this PR is merged I can discuss with Matt and try to add an additional output in this feedstock for Python users.)

**Comment by leofang** [comment]
cc: @conda-forge/cuda (in case anyone has bandwidth to chime in)

**Comment by leofang** [comment]
@conda-forge/cutlass shall we merge? Once this is merged I can get the lined-up updates done one by one.

**Comment by bdice** [comment]
@leofang Thanks! Responding to the CUDA team ping — yes, this looks great. If it seems appropriate, we could consider adding the conda-forge/cuda team as maintainers on this feedstock, like we did for CUDA Toolkit feedstocks.

**Comment by leofang** [comment]
> If it seems appropriate, we could consider adding the conda-forge/cuda team as maintainers on this feedstock, like we did for CUDA Toolkit feedstocks.

Sounds like a good idea, I suggest we discuss in the next sync-up. Once we are on board we can make the maintainer change any time, doesn't have to be part of this PR.

File: recipe/bld.bat
Comment (jakirkham) [inline recipe/bld.bat:31]: Looks like CMake might not install these. So this may be unneeded

If it is needed, probably we want this

```suggestion
rmdir /s /q %LIBRARYPREFIX%\test
```

File: conda-forge.yml
Comment (jakirkham) [inline conda-forge.yml:8]: Are we able to cross-compile for ARM too?

Then we could use Azure for all builds

```suggestion
  linux_aarch64: linux_64
```

File: recipe/meta.yaml
Comment (jakirkham) [inline recipe/meta.yaml:12]: Looks like we jumped from `0` to `2`. Suggesting `1` below. Though please let me know if I've missed something

```suggestion
  number: 1
```

File: recipe/meta.yaml
Comment (jakirkham) [inline recipe/meta.yaml:15]: As this is header only, should we drop this? Or is there is still a need to align CUTLASS versions between packages somehow?

```suggestion
```

**Review by jakirkham — COMMENTED** [review]
Thanks Leo! 🙏

Had a few questions and suggestions

Mainly moving to Ninja across the board

Also think there is a little more cleanup we can do now this is header only

All in the service of making this easier to maintain going forward

File: conda-forge.yml
Comment (leofang) [reply conda-forge.yml:8]: I think with this PR it becomes a simple recipe now so Travis is working fine, so I didn't bother to cross compile but either way is fine. No strong opinion. (Travis's ppc64le pipelines on the other hand are just not functioning.)

File: recipe/meta.yaml
Comment (leofang) [reply recipe/meta.yaml:17]: Yes we can, I was back and forth on this, and decided to only drop the most critical one (`cuda`) and keep the rest.

File: recipe/meta.yaml
Comment (leofang) [reply recipe/meta.yaml:15]: I think it's a good point. Happy to simplify it!

File: recipe/meta.yaml
Comment (leofang) [reply recipe/meta.yaml:12]: I swear I saw a `1` there when I started... lol

File: conda-forge.yml
Comment (jakirkham) [reply conda-forge.yml:8]: Personally would rather replace it

At this stage I'm unable to use Travis CI's web app at all. Have an open issue with them that has yet to be resolved after weeks

Would rather use cross-compilation for all of them if possible

Though please let me know if there is some reason to still use Travis here

Comment (leofang) [reply conda-forge.yml:8]: I will do anything I can to accelerate merging this PR, so I am accepting with no objection 🙂

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Review by jakirkham — APPROVED** [review]
Thanks Leo! 🙏

Approving

Maybe one of the maintainers can do a final pass and merge?

**Comment by leofang** [comment]
Thanks, John! @h-vetinari this is ready :)

**Comment by jakirkham** [comment]
Thanks all! 🙏

How would we like to handle updating the version here?

Do we want to just go through these in order?

* https://github.com/conda-forge/cutlass-feedstock/pull/14
* https://github.com/conda-forge/cutlass-feedstock/pull/19

Or do we just want to jump to latest?

**Comment by leofang** [comment]
Yes I'll get to them in order after I get to my desk.

**Comment by jakirkham** [comment]
No worries. Not trying to add pressure 😌

Just trying to understand how we want to approach updating 🙂

**Pull request merged**
